adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -725,7 +725,7 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
    * @return Returns a new GenomicRDD containing only data that overlaps the
    *   querys region.
    */
-  def filterByOverlappingRegions(querys: Iterable[ReferenceRegion]): U = {
+  def filterByOverlappingRegions(querys: java.lang.Iterable[ReferenceRegion]): U = {
```

Comment:
@heuermh we needed to change this to be accessible to python (py4j takes a python iterator and converts it to java ArrayList > java.lang.Iterable). Is there a better way to do this?

Reply:
The pattern we've been using so far is to have two methods, one marked as `Java-friendly`.  I'd recommend that here, unless the Scala or Java compiler isn't smart enough to distinguish the two different versions.

Reply:
@heuermh can you give me an example of this so I can copy the style you have been using?

Reply:
GenomicRDD has a Java friendly transmute method, https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala#L221 and others, just search for "friendly"  ;)